### PR TITLE
[Merged by Bors] - Set safe block hash to justified

### PR DESCRIPTION
**Pull request by michaelsproul**
## Issue Addressed

Closes https://github.com/sigp/lighthouse/issues/3189.

## Proposed Changes

- Always supply the justified block hash as the `safe_block_hash` when calling `forkchoiceUpdated` on the execution engine.
- Refactor the `get_payload` routine to use the new `ForkchoiceUpdateParameters` struct rather than just the `finalized_block_hash`. I think this is a nice simplification and that the old way of computing the `finalized_block_hash` was unnecessary, but if anyone sees reason to keep that approach LMK.

### CODE REVIEW

File: beacon_node/execution_layer/src/lib.rs [hunk 1]
```diff
@@ -601,58 +607,61 @@ impl<T: EthSpec> ExecutionLayer<T> {
...
                     .api
                     .get_payload_v1::<T>(payload_id)
                     .await
                     .map(|full_payload| {
                         if f(self, &full_payload).is_some() {
-                            warn!(self.log(), "Duplicate payload cached, this might indicate redundant proposal attempts.");
+                            warn!(
+                                self.log(),
+                                "Duplicate payload cached, this might indicate redundant proposal \
+                                 attempts."
```
Comment (michaelsproul) [inline beacon_node/execution_layer/src/lib.rs:663]: Fixing this over-long line caused `rustfmt` to format the section above correctly

**Comment by michaelsproul** [comment]
I'm still thinking about the best way to test this. I've heard there's a Hive test incoming, so perhaps we should merge this PR as-is relying on Hive, and come back for a more thorough round of EL testing in a later PR.

**Review by paulhauner — APPROVED** [review]
Looks great! No suggestions from me :rocket:

**Comment by michaelsproul** [comment]
fixing the batch after that spurious failure

bors r+

**Comment by paulhauner** [comment]
Rebatching

bors r+

**Pull request closed without merging**